### PR TITLE
[Storage] Updated tests to support multiple SAS versions

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs [hunk 1]
```diff
@@ -532,7 +532,11 @@ private void EnsureState()
                     Resource = Constants.Sas.Resource.BlobVersion;
                 }
             }
-            Version = SasQueryParameters.DefaultSasVersion;
+
+            if (string.IsNullOrEmpty(Version))
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs:536]: This is 1 of 2 logic changes I made in customer-facing code.  This if check is present in Data Lake, File Shares, and Queues, but was not present in blob for some reason.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs [hunk 2]
```diff
@@ -2590,7 +2590,12 @@ private void SetNameFieldsIfNull()
         /// A <see cref="Exception"/> will be thrown if a failure occurs.
         /// </remarks>
         public virtual Uri GenerateSasUri(DataLakeFileSystemSasPermissions permissions, DateTimeOffset expiresOn) =>
-            GenerateSasUri(new DataLakeSasBuilder(permissions, expiresOn) { FileSystemName = Name });
+            GenerateSasUri(
+                new DataLakeSasBuilder(permissions, expiresOn)
+                {
+                    FileSystemName = Name,
+                    Version = StorageVersionExtensions.ToVersionString(ClientConfiguration.Version)
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs:2597]: This is the other customer-facing change.  Now, the SAS version will match the x-ms-version of the client that generated it.  This is necessary to pass the correct sas version into our tests.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs:2597]: I also made this change in other client's `GenerateSasUri()` methods.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs [hunk 3]
```diff
@@ -77,7 +77,7 @@ public class DataLakePathClient
         /// <summary>
         /// <see cref="DataLakeClientConfiguration"/>.
         /// </summary>
-        private readonly DataLakeClientConfiguration _clientConfiguration;
+        internal readonly DataLakeClientConfiguration _clientConfiguration;
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs:80]: TODO undo this.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs]: Fixed.

File: sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs:536]: This was conscious change here https://github.com/Azure/azure-sdk-for-net/pull/15972 . Which means we forgot other places earlier. @amnguye please take a look.

**Pull request closed without merging**